rinja_parser/src/node.rs

```diff
@@ -167,6 +198,48 @@ impl<'a> Node<'a> {
     }
 }
 
+fn cut_node<'a, O>(
+    kind: Option<&'static str>,
+    inner: impl FnMut(&'a str) -> ParseResult<'a, O>,
+) -> impl FnMut(&'a str) -> ParseResult<'a, O> {
+    let mut inner = cut(inner);
+    move |i: &'a str| {
+        let result = inner(i);
+        if let Err(nom::Err::Failure(err) | nom::Err::Error(err)) = &result {
+            if err.message.is_none() {
```

Comment:
I can't wait for the `if let chain` feature to be stable...

Reply:
Yeah, I would like to remove a lot of `let Some(_) = f() else { return; }`*5 instances, too.